src/printer.rs

```diff
@@ -107,17 +110,57 @@ impl<'a, T: Read> BinaryGuard<'a, T> {
     }
 }
 
+pub struct FormatOptions {
+    pub json_indent: usize,
+    pub json_record_separator: String
+}
+
+impl FormatOptions {
+    pub fn from_string(options: Option<String>) -> FormatOptions {
```

Comment:
We usually implement `FromStr` trait for data types so the error can be reported by clap. See `Print` in `src/cli.rs` as an example.